### PR TITLE
Rails 5  + Open SSL version 2 support

### PR DESCRIPTION
**Pull request by jhirbour**
This PR adds to the work done by @nickcoyne to support Rails 5. In addition to Rails 5 support this Gem will now support Ruby versions 2.4 and greater. Ruby's Open SSL library changed in ruby 2.4.

### CODE REVIEW

**Comment by alexdean** [comment]
👏 👏 👏 👏 👏 👏 👏 👏 👏 👏 👏 👏 👏

File: Rakefile [hunk 1]
```diff
@@ -25,7 +25,7 @@ end
 namespace :admin do
   task :build do
     Dir.chdir('admin') do
-      sh 'ember build  --environment=production'
+      sh 'ember build production'
```
Comment (alexdean) [inline Rakefile]: will this require a new minimum version of `ember-cli`? if so, where/how will that be noted?

Comment (jhirbour) [reply Rakefile]: @romulomachado can you me with the correct bits on this? I changed ^ that per our chat. Where should I list the version of ember-cli ? I see `admin/bower.json`, but I'm unsure if that's the right place.

**Comment by jhirbour** [comment]
Just a note: during manual testing, I've discovered the changes I've made for SSL V2 don't work for DSA ssh keys. I'm troubleshooting it, and will fix/spec the change.

**Comment by jhirbour** [comment]
Ok so it turns out this app never supported DSA keys properly. This is mostly because of the `PubKey#very` . I investigated migrating this Gem to use [the ssh key gem](https://github.com/bensie/sshkey), but it doesn't support converting RSA keys to x509 Certs.

As such I'm shipping V1.0.0 of this gem as RSA keys work as expected.

File: Rakefile (hunk 1 above)
Comment (romulomachado) [reply Rakefile]: @jhirbour @alexdean This is more about `ember-cli-deploy` than `ember-cli`. 

Older versions of `ember-cli-deploy` won't work without the `--environment` flag, talkstar-frontend still uses an old version. 

Newer versions work fine with what's being changed here, donkey-frontend and programmer-frontend use a more recent version.

**Comment by jhirbour** [comment]
I squashed all the commits that were travis troubleshooting. I'm 🚢 ing this now.

**Pull request merged**